partiql-parser/src/partiql.pest

```diff
@@ -1,9 +1,13 @@
 WHITESPACE = _{ " " | "\t" | "\x0B" | "\x0C" | "\r" | "\n" }
 
-// really basic rules to just detect a sequence of keywords
-// two dip our toes in Pest
+// TODO implement a full grammar, this is a very primitive version to start
+//      working with Pest and its APIs.
 
-Keywords = _{ SOI ~ Keyword+ ~ EOI}
+// Entry point for full query parsing
+Query = _{ SOI ~ Keyword+ ~ EOI}
+
+// Entry point for query "scanning"
+Scanner = _{ SOI ~ Keyword }
```

Comment:
Two different "starting" rules, we use `SOI` here to indicate start of input (to also trigger white space parsing).  Notice that the `Scanner` rule does not have repetitions or `EOI`, the idea is to make it more of an "iterator" style API.

Reply:
maybe added the iterator bit to the comment